modules/oe_content_publication/oe_content_publication.module

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * @file
+ * OE Content Publication module file.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter() for the node form.
+ */
+function oe_content_publication_form_node_form_alter(array &$form, FormStateInterface $form_state) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $form_state->getBuildInfo()['callback_object']->getEntity();
+  if ($node->bundle() !== 'oe_publication') {
+    return;
+  }
+
+  $form['oe_documents']['#states'] = [
```

Comment:
Have you tried code
```
  $form['oe_documents']['widget'][0]['target_id']['#states'] = [
    'required' => [
      'input[name="oe_publication_collection"]' => ['value' => '0'],
    ],
  ];
  $form['oe_publication_publications']['widget'][0]['target_id']['#states'] = [
    'required' => [
      'input[name="oe_publication_collection"]' => ['value' => '1'],
    ],
  ];
```
It enables js required hint 
<img width="773" alt="Screenshot 2021-09-09 at 15 25 40" src="https://user-images.githubusercontent.com/1183951/132685684-d1e3b9cb-0b6f-47a5-b898-db266d9ad5e8.png">

since 
```
  $form['oe_documents']['#states'] = [
    'visible' => [
      'input[name="oe_publication_collection"]' => ['value' => '0'],
    ],
  $form['oe_publication_publications']['#states'] = [
    'required' => [
      'input[name="oe_publication_collection"]' => ['value' => '1'],
    ],
```
actually doesn't work.

Then we can avoid of using `_oe_content_publication_node_validate`

Reply:
I theme there is a notice error with the admin theme using the method above [0]['target_id'], anyhow we use validation in other places as well because it works even without JS.